modules/sdk-core/src/account-lib/mpc/shamir.ts

```diff
@@ -17,26 +16,47 @@ export default class Shamir {
    * @param secret secret to split
    * @param threshold share threshold required to reconstruct secret
    * @param numShares total number of shares to split to split secret into
-   * @param indices
+   * @param indices optional indices which can be used while generating the shares
+   * @param salt optional salt which could be used while generating the shares
    * @returns Dictionary of shares. Each key is an int in the range 1<=x<=numShares
    * representing that share's free term.
    */
-  split(secret: bigint, threshold: number, numShares: number, indices?: Array<number>): Record<number, bigint> {
+  split(
+    secret: bigint,
+    threshold: number,
+    numShares: number,
+    indices?: Array<number>,
+    salt = BigInt(0)
+  ): Record<number, bigint> {
     let bigIndices: Array<bigint>;
     if (indices) {
-      bigIndices = indices.map((i) => BigInt(i));
+      bigIndices = indices.map((i) => {
+        if (i < 1) {
+          throw new Error('Invalid value supplied for indices');
+        }
+        return BigInt(i);
+      });
     } else {
       // make range(1, n + 1)
       bigIndices = Array(numShares)
         .fill(null)
         .map((_, i) => BigInt(i + 1));
     }
-    assert(threshold > 1);
-    assert(threshold <= numShares);
+    if (threshold < 2) {
+      throw new Error('Threshold cannot be less than two');
+    }
+
+    if (threshold > numShares) {
+      throw new Error('Threshold cannot be greater than the total number of shares');
+    }
+
     const coefs: bigint[] = [];
     for (let ind = 0; ind < threshold - 1; ind++) {
-      const coeff = clamp(
-        bigIntFromBufferLE(crypto.createHmac('sha256', ind.toString(10)).update(bigIntToBufferLE(secret, 32)).digest())
+      const coeff = bigIntFromBufferLE(
```

Comment:
Just want to make sure here clamp doesn't bring any additional benefit as mentioned on the ticket, The eddsa tests have became flaky

Reply:
Are you saying that removing the clamp has made the tests flaky?

Reply:
This is the only significant change done to the code, I am getting intermittent CI failures with eddsa signing failing, not able to reproduce locally though @johnoliverdriscoll

Reply:
Ok I'll see if I can reproduce... it's been a while since I worked on the SDK, what commands are you using to test locally?

Reply:
I simply run the individual test case using vscode / intellij

Reply:
Which test(s) are failing intermittently?

Reply:
@johnoliverdriscoll https://github.com/BitGo/BitGoJS/runs/7501100144?check_suite_focus=true

Reply:
Looking at this again, I have problems with how the issue is written it the report. It suggests that clamping is not necessary and the alternative is to use a random 32 byte buffer, however, that would produce elements that are not in the Ed25519 field. The actual alternative would be to use a curve reduction, which is arguably less performant than clamping. I'm noting performance here because that is the only benefit cited in the issue. I suggest we continue to clamp. This will resolve the flaky tests.